### ISSUE TITLE
Unable to decrypt mails

### ISSUE THREAD

**Issue by mkapra** [mailcow/mailcow-dockerized#2395]
**Describe the bug, try to make it reproducible**
I moved my mailserver to another server. So I first made a backup of everything with the backup script. On the new Sever I installed mailcow regularly and restored the backup. After that no mails are coming anymore.

**System information and quick debugging**
General logs:
Follow lines are in my dovecot log:
`imap(mymailaddress)<182><yfy7hkmD+fld/sOm>: Error: Mailbox INBOX: UID=2940: read() failed: read(/var/vmail/mydomain/maximilian/cur/1551461809.M278114P1017.mail.mydomain.de,S=21519,W=22027:2,S) failed: Decryption error: no private key available (read reason=11/11 headers not cached (first=BCC) (Mail has other cached fields, reset_id=1532377773))
`
Further information (where applicable):
 - Ubuntu 18.04 LTS
 - KVM
 - 8 cores 30 GB RAM
 - Output of `git diff origin/master`, any other changes to the code? -> no output

**Comment by hachre** [comment]
Was your Mailcow quite up-to-date before you did the backup and move? Do you still have access to the old server? What seems to be missing is the contents of the "crypt" part of the backup. But if your installation was very outdated, it wasn't used yet for mail encryption and the restore to a much newer version (without updating to it) may have screwed things up.

**Comment by andryyy** [comment]
It would read unencrypted files just fine. It only tries to decrypt files with a CRYPT(ED?) header.

Check your `/var/lib/docker/volumes/mailcowdockerized_crypt-vol-1/_data/` directory and permissions of the ec*pem files. What do they look like?

**Comment by gummipunkt** [comment]
IRC quote :)
-rw------- 1  401 root  241 Sep  8 17:59 ecprivkey.pem-rw-r--r-- 1  401 root  178 Sep  8 17:59 ecpubkey.pem

And I think "root" is the problem

**Comment by andryyy** [comment]
Looks good.

I would try to restore vmail only again.

**Comment by andryyy** [comment]
If it’s important, contact me, please. :)

**Comment by Triscuit47** [comment]
Hey guys, any follow-up on this issue? I ran into the same thing after restoring a month-old backup.

**Comment by florian-g-** [comment]
Same here! Any news? 

File permissions in /var/lib/docker/volumes/mailcowdockerized_crypt-vol-1/_data/ directory look the same as above and restoring only vmail does not make any difference either.
Strange. Any idea?

**Comment by andryyy** [comment]
Both mailcows need to be the exact same patch level for this to work. 

Perhaps you need to set `MAILDIR_SUB=` in mailcow.conf. That's all support I can offer here. Check the Telegram channel. Perhaps someone can get their hands on this?

**Comment by Triscuit47** [comment]
@florian-g- a workaround I did for this was to set up a temporary mailcow with the old backup, decrypt my old emails there and then imap sync them to the current build.

**Comment by BerFoe** [comment]
I just ran into this issue after migration my installation. All mailboxes i've checked seemed to be ok, except one. I tried several backups, updated a second time on both systems to be sure, that patch level is exactly the same. But it just did not help.
Finally I just deleted those files directly out of the vmail-folder, it were just three mails (got the path from dovecot-log). Just after that, everything works again. Sadly I don't know how to find out which mails exactly I needed to delete, but I think a one-time sync job should be fine.
